### DISCUSSION TITLE
Replay attacks of proof-authorised account updates across multiple networks

### DISCUSSION THREAD

**Discussion by MartinOndejka** [MinaProtocol/mina#13689] [Ideas]
In the [zkapp MIP](https://github.com/MinaProtocol/MIPs/blob/main/MIPS/mip-zkapps.md#transaction-replay-attacks), the replay attacks measures mostly revolve around correct nonce. Currently to prevent replaying of signed commands from berkeley to mainnet, there is a custom salt for hashing, which will create unique signature for any network. This will work also for signature-authorised account updates because of the unique signature, but what about the proof-authorised account updates? The nonce precondition can be bypassed easily. Does the `use_full_commitment` play role in the proof in some way so the fee payer hash (with network salt) is included? If not is there some another plan to prevent replaying of berkeley account updates on mainnet after hardfork?

**Comment by mitschabaude** [comment]
> The nonce precondition can be bypassed easily.

How? Can you elaborate on that?

**Reply by mitschabaude** [reply]
Oh I think you mean it doesn't prevent replay across networks, right?

**Reply by MartinOndejka** [reply]
Yeah, for example replaying berkeley account updates on mainnet (or maybe some new network). Afaik the hash salt is in place to prevent from replaying payments that way.

**Comment by mrmr1993** [comment]
> Does the `use_full_commitment` play role in the proof in some way so the fee payer hash (with network salt) is included?

No.

> If not is there some another plan to prevent replaying of berkeley account updates on mainnet after hardfork?

No. The easiest way would be to add domain separation for the [account update hash prefix](https://github.com/MinaProtocol/mina/blob/f20f341add3c2285a06f0dadbf8dce040905586b/src/lib/hash_prefixes/hash_prefixes.ml#L91), which would be exactly analogous to [what we do for signatures](https://github.com/MinaProtocol/mina/blob/develop/src/lib/hash_prefixes/hash_prefixes.ml#L47-L51)

**Reply by MartinOndejka** [reply]
> The easiest way would be to add domain separation for the [account update hash prefix](https://github.com/MinaProtocol/mina/blob/f20f341add3c2285a06f0dadbf8dce040905586b/src/lib/hash_prefixes/hash_prefixes.ml#L91)

Thanks, that sounds like a good idea. Just to get the better understanding, how is account update hash being used?

**Reply by mitschabaude** [reply]
The account update hash is the public input to the proof! So proofs wouldn't verify on a different network
